gossipd/broadcast.c

```diff
@@ -22,6 +22,29 @@ static struct queued_message *new_queued_message(tal_t *ctx,
 	return msg;
 }
 
+bool replace_broadcast(struct broadcast_state *bstate, u64 *index,
+		       const int type, const u8 *tag, const u8 *payload)
+{
+	struct queued_message *msg;
+	bool evicted = false;
+
+	msg = uintmap_get(&bstate->broadcasts, *index);
+	if (msg && msg->type == type && memcmp(msg->tag, tag, tal_len(tag)) == 0) {
```

Comment:
Are tags always the same size? What happens if the `msg->tag` is not same size as `tag`?

Reply:
They are now, were not previously and I patched it :)